src/main/java/au/org/emii/netcdf/iterator/reader/NetcdfReader.java

```diff
@@ -11,15 +11,19 @@
 import ucar.ma2.ArrayChar;
 import ucar.ma2.Index;
 import ucar.nc2.Variable;
+import ucar.nc2.dataset.EnhanceScaleMissing;
 
 public class NetcdfReader {
 
     private Array data;
     
     private IndexValuesConverter indexValuesConverter;
     private Index index;
+    private final Variable variable;
 
     public NetcdfReader(Variable variable) throws IOException {
```

Comment:
Not very nice to have the constructor potentially throwing an IO exception.

Would be better if it just did simple assignment so it can't fail.

Then in the accessors do ``` variable.read() ``` stuff.

I'd expect IO errors on actual  reading values.